### ISSUE TITLE
[Feature] Override the serializer for json

### ISSUE THREAD

**Issue by ShravanSunder** [honojs/hono#1706]
### What is the feature you are proposing?

It would be great to be able to easily override the serializer in req/res of .json.  I would like to use superjson to make date handling easier.

**Comment by yusukebe** [comment]
Hi @ShravanSunder 

Honestly, I don't want to add more features to `c.json()`, even though the changes might not be big.

Instead, you can use `c.render()` and `c.setRenderer()`:

https://hono.dev/api/context#render-setrenderer

**Comment by ShravanSunder** [comment]
@yusukebe Just so that i understand well, i would set my custom renderer with setRender and use that?  I think that's a good fallback.  Would this also work with hono stacks?

**Comment by yusukebe** [comment]
@ShravanSunder 

Yes. You can write like the following though it will be a little bit verbose.

```ts
import { Hono } from 'hono'

const app = new Hono()

declare module 'hono' {
  interface ContextRenderer {
    (data: object): Response
  }
}

app.get('/api/*', async (c, next) => {
  c.setRenderer((data) => {
    return c.body(JSON.stringify(data, null, 4))
  })
  await next()
})

app.get('/api/foo', (c) => {
  return c.render({
    foo: 'bar',
  })
})
```

**Comment by ShravanSunder** [comment]
@yusukebe thanks!

One more question, is this method also possible to use with https://hono.dev/concepts/stacks

**Comment by yusukebe** [comment]
Ah, you want to use `jsonT()`?

**Comment by yusukebe** [comment]
There might be someone using `superjson` for RPC, but I can't recall it now. Hmm.

**Comment by ShravanSunder** [comment]
i think most specifically the handling of dates is the best thing out of superjson.  Makes things uncomplicated

**Comment by jvhellemond** [comment]
FWIW: I'm going to try the same (using `setRenderer()` and `render()`) to be able to "support" `Infinity` and `-Infinity` in JSON.

**Comment by noahehall** [comment]
hmm i'm going to try this with [msgpack](https://github.com/msgpack/msgpack-javascript)

**Comment by maou-shonen** [comment]
I also hope to see this feature.
Although c.render can work well,
I believe using `c.json` for handling JSON and `c.render` for handling other more complex things would be more intuitive.
~~It would also prevent me from accidentally using the wrong method in different repository~~

**Comment by musjj** [comment]
I would love to see this too. AFAIK `setRenderer()` can be used to deserialize the data to a superjson string, but you still need to serialize it manually on the RPC client side right?
